nomenclature/code.py

```diff
@@ -233,13 +233,27 @@ class RegionCode(Code):
         Name of the RegionCode
     hierarchy : str
         Hierarchy of the RegionCode
-    iso3_codes : str or list of str
+    countries : list of str, optional
+        List of countries in that region
+    iso3_codes : str or list of str, optional
         ISO3 codes of countries in that region
 
     """
 
     hierarchy: str = None
-    iso3_codes: Union[List[str], str] = None
+    countries: Optional[List[str]] = None
+    iso3_codes: Optional[Union[List[str], str]] = None
+
+    @field_validator("countries")
+    def check_countries(cls, v: List[str], info: ValidationInfo) -> List[str]:
+        """Verifies that each country name is defined in `nomenclature.countries`."""
+        if invalid_country_names := set(v) - set(countries.names):
+            raise ValueError(
+                f"Region '{info.data['name']}' uses non-standard country name(s): "
```

Comment:
I'd reference the exact ISO norm that we're using, just "standard" is a little ambiguous in my opinion:

```suggestion
                f"Region '{info.data['name']}' uses non-ISO3166-1 compliant country name(s): "
```

Reply:
We do not strictly follow ISO-3166-1, we use our own adaptation, see 
https://github.com/IAMconsortium/nomenclature/blob/main/nomenclature/countries.py.

I added a link to the documentation for clarity.